### ISSUE TITLE
6.3.1 Swashbuckle.AspNetCore.Cli  Instances cannot be resolved and nested lifetimes cannot be created exception

### ISSUE THREAD

**Issue by slnkykrn** [domaindrivendev/Swashbuckle.AspNetCore#2452]
Hello ,
We're using cli tool like below in our csproj .net 6 framework : 

 <Target Name="SwaggerToFile" AfterTargets="AfterBuild">
    <Exec Command="dotnet $(DotNetSwaggerPath) tofile --host http://localhost:51071 --output wwwroot/swagger/v1/swagger.json $(OutputPath)$(AssemblyName).dll v1" />
  </Target>

But we're getting an error like below , It depends on usage of Autofac I think. 

![image](https://user-images.githubusercontent.com/3687733/176114751-e1ee966e-1cc1-4de2-9410-b73eeac68bf8.png)

When I add below code : We'll be getting an error this time  

public class SwaggerHostFactory
{
    public static IHost CreateHost()
    {
        return Program.CreateHostBuilder(new string[0]).Build();
    }
}

error after hostFactory : 
![image](https://user-images.githubusercontent.com/3687733/176116555-30ba8a32-6d1d-4f85-917b-f26d0126016e.png)

Can you help us about this issue ? 
Thank you.

**Comment by slnkykrn** [comment]
Is there any idea ? 
Thank you

**Comment by homemdasneves** [comment]
I'm also facing this issue. I was having no problems at all until I added Hangfire to my project. I was able to narrow it down to this line: 
`// Add the processing server as IHostedService
        services.AddHangfireServer();`

If I remove it, the error goes away.

I don't need hangfire when I'm running the Swashbucle Cli, so a basic workaround in this specific case, would be to disable hangfire when running inside Cli. Does anybody know how to detect that?
